.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,160 @@
+
+`zdb` utility is the main tool to debug ZFS. 
+
+Additionally, see 
+[build instruction](https://github.com/zfsonlinux/zfs/wiki/Building-ZFS)
+, there is `--enable-debug` flag, which can be set to enable all ASSERTs and additional correctness tests.
+
+### Where can I ask for help?
+The best place to ask is our [mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists).
+
+Please *note*, our issue tracker is **only for bugs**, if you have thoughts about any 
+enhancements - first tell about them via 
+[mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists).
+
+## How Can I Contribute?
+
+### Reporting Bugs
+
+You can open a new issue and search existing issues using the public
+[issue tracker](https://github.com/zfsonlinux/zfs/issues). 
+
+Please make sure that what you're actually seeing is a bug and **not a support issue**. 
+
+If in doubt, please ask on the [mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists) 
+first, and if you're then asked to file an issue, do so.
+
+####When opening a new issue include this information at the top of the issue:
+
+* What distribution you're using and the version.
+* What spl/zfs packages you're using and the version.
+* Describe the problem you're observing.
+* Describe how to reproduce the problem.
+* Including any warning/errors/backtraces from the system logs.
+
+When a new issue is opened it's not uncommon for a developer to request additional 
+information about the problem. 
+
+In general, the more detail you share about a problem the quicker a developer 
+can resolve it. For example, providing a simple test case is always 
+exceptionally helpful.  
+
+Be prepared to work with the developer looking in to your bug in order to get 
+it resolved. They may ask for information like:
+
+* Your pool configuration as reported by `zdb` or `zpool status`.
+* Your hardware configuration, such as
+  * Number of CPUs.
+  * Amount of memory.
+  * Whether your system has ECC memory.
+  * Whether it is running under a VMM/Hypervisor.
+  * Kernel version.
+  * Values of the spl/zfs module parameters.
+* Stack traces which may be logged to `dmesg`.
+
+####Issue template
+
+```
+**Which distribution are you using (version, kernel)**
+
+[Distribution, version, kernel version, architecture]
+
+**Which versions of spl/zfs packages are you using**
+
+[zfs ver, spl ver]
+
+**Describe the problem you're observing**
+
+[Describe the problem]
+
+**Describe how to reproduce the problem**
+
+1. [First Step]
+2. [Second Step]
+3. [Other Steps...]
+
+**Include any warning/errors/backtraces from the system logs**
+
+Wrap log text with ```
+
+[any log, see https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#reporting-bugs]
+```
+
+### Suggesting Enhancements
+
+For now [mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists) is the best place
+to discuss any enhancements.
+
+ZFS on Linux is stable, and the main priorities for our team are cleaning 
```

Comment:
`We attempt to keep ZFS on Linux in a stable state. The current priorities are cleaning up the code, fixing any bugs, and improving performance.`? By saying `main priorities` it sounds like we aren't adding new features often. Thoughts?